primitives/io/src/lib.rs

```diff
@@ -370,6 +373,13 @@ pub trait Misc {
 			.call_in_wasm(wasm, "Core_version", &[], &mut ext)
 			.ok()
 	}
+
+	/// Get the current system time in nanoseconds.
+	fn system_time() -> u128 {
```

Comment:
Are we going to support the benchmark, whatever they called macros from within the runtime? Or is this just for the benchmark purposes?

Nevertheless, I want some big exclamation marks in the documentation of this function. The user needs to be aware that this should not be used for anything that touches the storage, as otherwise we get storage root mismatches.

Reply:
```suggestion
	fn current_time() -> u128 {
```

Reply:
It should be for benchmarking purposes only. I can add more documentation.